linked-lists-python/linked-lists-python.py

```diff
@@ -0,0 +1,92 @@
+class Node:
+    def __init__(self, data):
+        self.data = data
+        self.next = None
+
+    def __repr__(self):
+        return self.data
+
+
+class LinkedList:
+    def __init__(self, nodes=None):
+        self.head = None
+        if nodes is not None:
+            node = Node(data=nodes.pop(0))
+            self.head = node
+            for elem in nodes:
+                node.next = Node(data=elem)
+                node = node.next
+
+    def __repr__(self):
+        node = self.head
+        nodes = []
+        while node is not None:
+            nodes.append(node.data)
+            node = node.next
+        nodes.append("None")
+        return " -> ".join(nodes)
+
+    def __iter__(self):
+        node = self.head
+        while node is not None:
+            yield node
+            node = node.next
+
+    def add_first(self, node):
+        node.next = self.head
+        self.head = node
+
+    def add_last(self, node):
+        if not self.head:
+            self.head = node
+            return
+        for current_node in self:
+            pass
+        current_node.next = node
+
+    def add_after(self, target_node_data, new_node):
+        if not self.head:
+            raise Exception("List is empty")
+
+        for node in self:
+            if node.data != target_node_data:
+                continue
+            new_node.next = node.next
+            node.next = new_node
```

Comment:
Just the same comment here that I made in the article - if you invert the logic of the if statement, you can remove the continue. :)